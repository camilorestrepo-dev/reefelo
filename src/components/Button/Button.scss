@import '../../styles/variables.scss';

.custom-btn {
  border: none;
  border-radius: $radius-5;
  font-size: medium;
  font-weight: bold;
  padding: $spacing-8 $spacing-12;

  &:hover {
    cursor: pointer;
  }

  &.solid-btn {
    color: $textColor;
    background-color: $primary;
  }

  &.outline-btn {
    color: $primary;
    background-color: $white;
    border: 1px solid $primary;
  }

  &.link-btn {
    color: $primary;
    background-color: $white;
    border: none;
  }
}