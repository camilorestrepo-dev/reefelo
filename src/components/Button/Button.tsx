import classnames from "classnames";
import "./Button.scss";

export type ButtonProps = {
  label: string;
  type?: "button" | "reset" | "submit";
  variant: "solid" | "outline" | "link";
  onClick?: VoidFunction;
};

const Button: React.FC<ButtonProps> = ({ label, type, variant, onClick }: ButtonProps) => (
  <button
    type={type || "button"}
    className={classnames("custom-btn", `${variant}-btn`)}
    onClick={onClick}
  >
    {label}
  </button>
);

export default Button;
