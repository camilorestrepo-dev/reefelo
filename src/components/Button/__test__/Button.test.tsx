import { render } from "@testing-library/react";
import Button, { ButtonProps } from "../Button";

const props: ButtonProps = {
  label: "Some button",
  variant: "solid",
};

test("render solid", () => {
  const component = render(<Button {...props} />);
  expect(component.container).toMatchSnapshot();
});

test("render outline", () => {
  const newProps: ButtonProps = { ...props, variant: "outline" };
  const component = render(<Button {...newProps} />);
  expect(component.container).toMatchSnapshot();
});

test("render link", () => {
  const newProps: ButtonProps = { ...props, variant: "link" };
  const component = render(<Button {...newProps} />);
  expect(component.container).toMatchSnapshot();
});
