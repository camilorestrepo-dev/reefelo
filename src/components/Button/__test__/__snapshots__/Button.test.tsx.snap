// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render link 1`] = `
<div>
  <button
    class="custom-btn link-btn"
    type="button"
  >
    Some button
  </button>
</div>
`;

exports[`render outline 1`] = `
<div>
  <button
    class="custom-btn outline-btn"
    type="button"
  >
    Some button
  </button>
</div>
`;

exports[`render solid 1`] = `
<div>
  <button
    class="custom-btn solid-btn"
    type="button"
  >
    Some button
  </button>
</div>
`;
